lib/Syntax/SyntaxNodes.cpp.gyb

```diff
@@ -75,26 +75,30 @@ ${child.type_name} ${node.name}::get${child.name}() const {
 %         raise Exception("'collection_element_name' should be set for '%s' of '%s'" % (child.name, node.name))
 %       end
 ${node.name} ${node.name}::add${child_elt}(${child_elt_type} ${child_elt}) {
-  RC<RawSyntax> raw = getRaw()->getChild(Cursor::${child.name});
+  const RawSyntax *raw = getRaw()->getChild(Cursor::${child.name});
   if (raw)
     raw = raw->append(${child_elt}.getRaw());
   else
     raw = RawSyntax::makeAndCalcLength(SyntaxKind::${child_node.syntax_kind},
-      {${child_elt}.getRaw()}, SourcePresence::Present);
+      {${child_elt}.getRaw()}, SourcePresence::Present, raw->getArena());
   return ${node.name}(Data.replacingChild(raw, Cursor::${child.name}));
 }
 %     end
 
 ${node.name} ${node.name}::with${child.name}(
   llvm::Optional<${child.type_name}> New${child.type_name}) {
-  RC<RawSyntax> raw;
+  const RawSyntax *raw;
   if (New${child.type_name}.hasValue()) {
     raw = New${child.type_name}->getRaw();
   } else {
 % if child.is_optional:
     raw = nullptr;
 % else:
-    raw = ${make_missing_child(child)};
+    {
+      // make_missing_child access the 'Arena' variable. Create it.
+      const RC<SyntaxArena> &Arena = getRaw()->getArena();
```

Comment:
It used to be a reference to avoid ref-counting the arena. But of course it shouldn’t be since `RawSyntax::getArena` is no longer returning a reference. Good catch.